std/machines/small_field/pointer_arith.asm

```diff
@@ -0,0 +1,35 @@
+/// Creates constraints that increments a pointer by one full 32-bit word.
+///
+/// The pointers are given in 2 16-bit limbs. This function return the polynomials constraining
+/// (int(post_high) * 0x10000 + int(post_low)) to (int(pre_high) * 0x10000 + int(pre_low) + 4).
+///
+/// The returned constraints are not automatically added to the constraints set. This is so the
+/// caller will be able to latch-disable them in rows they aren't needed.
+///
+/// This constr function introduces 2 new witness column and some helper constraints.
+///
+/// Preconditions: pre_low must be 16-bit aligned and in range [0, 0xfffc].
+let word_increment_ptr: expr, expr, expr, expr -> Constr[] = constr |pre_high, pre_low, post_high, post_low| {
+    // How far away from overflowing the low limb is:
+    let low_diff = pre_low - (0x10000 - 4);
+    // Is one if low limb is about to overflow:
+    let low_overflow;
+    // Helper to allow low_overflow to be boolean
+    let low_diff_inv;
+
+    low_overflow = 1 - low_diff_inv * low_diff;
```

Comment:
The constraints here ensure that `low_overflow` is `1` if and only if `low_diff` is zero, otherwise it is `0`, right? Should we also extract that into a helper function? Then we can use it for `XIsZero`.

Reply:
Yes. Just found `utils::is_zero()`.

Reply:
Done.